Dsikora369_code/task_1/titanic.py

```diff
@@ -0,0 +1,286 @@
+    age_null_random_list = rng.uniform(age_avg - age_std, age_avg + age_std,
+                                       size=age_null_count)
+    dataset['Age'][np.isnan(dataset['Age'])] = age_null_random_list
+
+    # binning with cut
+    dataset['Age'] = pd.cut(dataset['Age'], 5)
+
+    # apply regex
+    dataset['Title'] = dataset['Name'].apply(get_title)
+    # replace
+    dataset['Title'] = dataset['Title'].replace(['Lady', 'Countess', 'Capt',
+                                                 'Col', 'Don', 'Dr', 'Major',
+                                                 'Rev', 'Sir', 'Jonkheer',
+                                                 'Dona'], 'Rare')
+    # replace
+    dataset['Title'] = dataset['Title'].replace(['Mlle', 'Ms'], 'Miss')
+    # replace
+    dataset['Title'] = dataset['Title'].replace('Mme', 'Mrs')
+    # fill nans
+    dataset['Title'] = dataset['Title'].fillna(0)
+    # fill nans
+    dataset['Cabin'] = dataset['Cabin'].fillna('Z')
+    # apply replacement
+    dataset['BL'] = str(dataset['Cabin'].apply(get_cabin))
+
+    # drop columns
+    drop_elements = ['PassengerId', 'Name', 'Ticket', 'SibSp',
+                     'Parch', 'FamilySize', 'Cabin']
+    dataset = dataset.drop(drop_elements, axis=1)
+
+    # encode labels
+    le = LabelEncoder()
+
+    le.fit(dataset['Sex'])
+    dataset['Sex'] = le.transform(dataset['Sex'])
+
+    le.fit(dataset['Title'])
+    dataset['Title'] = le.transform(dataset['Title'])
+
+    le.fit(dataset['Embarked'].values)
+    dataset['Embarked'] = le.transform(dataset['Embarked'].values)
+
+    le.fit(dataset['BL'])
+    dataset['BL'] = le.transform(dataset['BL'])
+
+    le.fit(dataset['Fare'])
+    dataset['Fare'] = le.transform(dataset['Fare'])
+
+    le.fit(dataset['Age'])
+    dataset['Age'] = le.transform(dataset['Age'])
+    # creating ageclass feature
+    dataset['Age*Class'] = dataset.Age * dataset.Pclass
+    le.fit(dataset['Age*Class'])
+    dataset['Age*Class'] = le.transform(dataset['Age*Class'])
+
+    return dataset
+
+
+dataset = load_data("train.csv")
+train, val = train_test_split(dataset, test_size=0.1)
+features = ['Pclass', 'Sex', 'Age', 'Fare', 'Embarked', 'Alone', 'Title',
+            'BL', 'Age*Class']
+training_dataset = (
+    tf.data.Dataset.from_tensor_slices(
+        (
+            tf.cast(train[features].values, tf.float32),
+            tf.cast(train['Survived'].values, tf.int32)
+        )
+    )
+)
+validation_dataset = (
+    tf.data.Dataset.from_tensor_slices(
+        (
+            tf.cast(val[features].values, tf.float32),
+            tf.cast(val['Survived'].values, tf.int32)
+        )
+    )
+)
+
+"""**Creating model**"""
+model = tf.keras.Sequential([
+    Dense(12, activation=tf.nn.leaky_relu, input_shape=[9]), Dropout(0.2), 
+    Dense(5, activation=tf.nn.leaky_relu), Dense(2, activation=tf.nn.softmax)])
+opt = tf.keras.optimizers.Adam(
+    learning_rate=0.01, beta_1=0.9, beta_2=0.999, epsilon=1e-07, amsgrad=True,

```

Comment:
Note that if you are using default function arguments you can left them unspecified 
https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam